src/main/java/com/rtbhouse/kafka/workers/impl/KafkaWorkersImpl.java

```diff
@@ -104,12 +105,12 @@ public void start() {
         // - plus one consumer thread
         // - plus one punctuator thread
         final int allThreadsNum = workerThreadsNum + 2;
-        executorService = Executors.newFixedThreadPool(allThreadsNum);
-        executorService.execute(consumerThread);
+        executor = (ThreadPoolExecutor) Executors.newFixedThreadPool(allThreadsNum);
```

Comment:
So maybe use `ThreadPoolExecutor::new`?

Reply:
Tak jest krócej i testy to pokrywają (a szansa, że się zwali jest minimalna).